src/TomKerkhove.Dapr.Streaming.Runtimes.AzureFunctions/Functions/AzureEventHubsFunction.cs

```diff
@@ -54,12 +55,13 @@ protected virtual async Task ProcessIndividualEventAsync(EventData eventData, Li
         {
             try
             {
-                if (eventData.Body == null || eventData.Body.Array == null)
+                if (eventData.Body.IsEmpty || eventData.Body.ToArray().Length == 0)
                 {
                     throw new Exception("Message does not contain a payload");
                 }
 
-                string rawEventPayload = Encoding.UTF8.GetString(eventData.Body.Array, eventData.Body.Offset, eventData.Body.Count);
+                string rawEventPayload = Encoding.UTF8.GetString(eventData.Body.Span);
+                //string rawEventPayload = Encoding.UTF8.GetString(eventData.Body.Array, eventData.Body.Offset, eventData.Body.Count);
```

Comment:
Let's remove the legacy
```suggestion
                string rawEventPayload = Encoding.UTF8.GetString(eventData.Body.Span);
```